### PR TITLE
Add v6 tests (WIP)

### PR DESCRIPTION
**Pull request by timwis**
Not ready to merge yet, work in progress. Jotted down ideas for tests below. Let me know if any don't make sense, and feel free to add other ideas.

## Unit test ideas
- [x] enables history by default
- [ ] if history enabled, registers navigate, popstate, pushstate, replacestate events
- [x] enables href by default
- [ ] if href enabled, clicking link trigers pushstate (failing, likely due to nanohref cancelling in node)
  - [ ] unless link is current location
- [ ] `.route` registers the route
- [x] route is passed state and emit fn
  - [ ] verify emit fn context?
- [x] state is passed current route
- [ ] state is passed current search query (jsdom doesn't support changing search)
- [x] callback passed to `.use` is passed state, emitter, and app
  - [ ] verify context?
- [ ] on navigate event, emits render
- [ ] on navigate event, updates query
- [ ] on navigate event, scrolls to anchor
- [ ] on popstate event, emits navigate
- [ ] on pushstate event, modifies history and emits navigate
- [ ] on replacestate event, replaces history and emits navigate
- [ ] on window.onpopstate, emits popstate event
- [ ] domcontentloaded event fired on document ready
- [ ] mount morphs the target node
- [ ] toString works
- [ ] on render event, tree is morphed

## higher level (closer to integration tests)
- [ ] state change re-renders? (ie. choo 101 app)
- [ ] `*` route called by default
  - [ ] `/*` route works too (maybe nanorouter)

### CODE REVIEW

**Comment by yoshuawuyts** [comment]
> is passed state, emitter, and app

Oops, that thould not contain `app` haha

**Comment by yoshuawuyts** [comment]
This is all great!

File: test.js [hunk 1]
```diff
@@ -1,9 +1,14 @@
 var tape = require('tape')
+var Nanobus = require('nanobus')
+require('jsdom-global')(null, { url: 'http://localhost/' })
+window.localStorage = window.localStorage || {} // See: yoshuawuyts/nanotiming#7
+window.requestAnimationFrame = window.requestAnimationFrame || function (cb) { process.nextTick(cb) }
```
Comment (yoshuawuyts) [inline test.js]: This looks a little tricky; perhaps using https://github.com/juliangruber/tape-run might make it easier by testing in an actual browser env?

Comment (timwis) [inline test.js]: Oh heck yeah, that'd make it easier. At least 1 other test simply wouldn't work otherwise too. Glad to hear you're good using that.

**Comment by yoshuawuyts** [comment]
@timwis do you think this could be merged?

**Comment by goto-bus-stop** [comment]
@timwis do you have time to rebase and merge soon? else would you mind if i do it? :stuck_out_tongue:

**Comment by timwis** [comment]
I'm sooooo sorry I missed your comments! I was subscribed to all activity on the repo and got behind reading every single comment, so didn't realise I was mentioned here. I started to rebase this but decided to start over in another branch with hopefully better tests. Just submitted another PR.

**Pull request closed without merging**